content/tutorials/quickstart/get-nftoken-list.md

```diff
@@ -0,0 +1,337 @@
+
+You can list all of the tokens that currently exist on the XRP Ledger. The process involves repeatedly using the `ledger_data` request, filtering the `NFToken` page objects, then listing the array of `NFToken` objects stored on the page. This example demonstrates how to extract `NFToken` objects and `NFTokenIDs`, and how to get information for the listed NFTokens using the `nft_info` request to the Clio server.
+
+## Usage
+
+You can download the [Quickstart Samples](https://github.com/XRPLF/xrpl-dev-portal/tree/master/content/_code-samples/quickstart/js/quickstart.zip){.github-code-download} archive to try the sample in your own browser.
+
+### Get NFTokens
+
+All transactions on the XRP Ledger are available for public review. You do not need an account or credentials to access information about minting and trading NFTokens.
+
+To get a list of current `NFTokens` on the XRP Ledger, click **Get NFTokens**. The button retrieves 200 transactions (the maximum batch allowance), then filters to show only the information from `NFToken` pages. The row marker for the response is stored in the **Marker** field. Each new search starts at the marker to return a unique set of records for the next 200 transactions. Click **Get NFTokens** repeatedly to retrieve all of the `NFTokens` on the XRP Ledger.
+
+### Get NFTokenIDs
+
+Click **Get NFToken IDs** for a more finely grained search that returns only the `NFTokenIDs` from each batch of 200 transactions. Continue to click **Get NFToken IDs**  to retrieve all `NFTokenIDs` on the XRP Ledger.
+
+
+### Get Info
+
+Having a list of `NFTokens` is all well and good, but it would be more interesting to do something with the `NFTokenIDs` returned. You can get information on the `NFTokens` in the response.
+
+To review `NFToken` information for a selected `NFTokenID`:
+
+1. Copy a `NFTokenID` from the list and paste it in the **NFTokenID** field.
+2. Click **Get NFToken Info**.
+
+
+# Code Walkthrough
+
+You can download the [Quickstart Samples](https://github.com/XRPLF/xrpl-dev-portal/tree/master/content/_code-samples/quickstart/js/quickstart.zip){.github-code-download} archive to try each of the samples in your own browser.
+
+## Get `NFTokens`
+
+Declare variables used in the function.
+
+```javascript
+async function getNFTokens () {
+	let jason
+	let ledger_entry_type
+	let nfTokens
+	let nftoken_data
+	let results = ""
+	let stateTree
+```
+
+Connect to the XLS-20Devnet server.
+
+```javascript
+  const client = new xrpl.Client('wss://xls20-sandbox.rippletest.net:51233')
+	await client.connect()
+```
+
+Check to see if the **Marker** field is populated with a 64 character value. If so, include the marker value in the `ledger_data` request. Set the limit of records to be returned to 200 transactions.
+
+```javascript
+  if (markerField.value.length == 64) {
+		nftoken_data = await client.request({
+			command: "ledger_data",
+			id: "NFToken data request",
+			limit: 200,
+			marker: markerField.value
+		})
```

Comment:
I notice this doesn't specify a ledger index / hash. I'm not 100% sure how `ledger_data` handles its markers, but I am worried this might mean that the ledger you're iterating over might shift while you're in the process of iterating it.

Probably the more secure way of doing this is specifying `ledger_index: "validated"` in the case where there's no marker, and specifying `ledger_index: nftoken_data.result.ledger_index` in the case where there _is_ a marker. (That way, you default to "whatever the latest validated ledger is" for the initial call, and "the ledger index you started with" for each follow-up.) I think technically with the way this script is architected you also have to do something extra to make sure that the response from the _previous_  `ledger_data` call (well, at least the `result.ledger_index` value) is available when you do the next call.